server/punishments.ts

```diff
@@ -954,16 +954,15 @@ export const Punishments = new class {
 		const punishment = ['BATTLEBAN', id, expireTime, ...reason] as Punishment;
 
 		// Handle tournaments the user was in before being battle banned
-		for (const games of user.games.keys()) {
-			const game = Rooms.get(games)!.getGame(Tournaments.Tournament);
-			if (!game) continue; // this should never happen
+		for (const room of user.getRooms()) {
```

Comment:
```suggestion
		for (const room of Rooms.rooms.values()) {
```

It's important to never forget that users can be in games without being in the game's room.